packages/modules-sdk/src/medusa-module.ts

```diff
@@ -262,14 +264,13 @@ export class MedusaModule {
     let finishLoading: any
     let errorLoading: any
 
+    const loadingPromise = new Promise((resolve, reject) => {
```

Comment:
Since we are in an async function, can't we just `return` or `throw` instead of this pattern?

Reply:
This prevents the same module being initialized multiple times while loading, like if you load it in a lambda handler and that is called multiple times in concurrency.
There isn't any other way to achieve this in a single process other than sharing a single Promise.